core/src/main/scala/org/apache/spark/network/netty/SparkTransportConf.scala

```diff
@@ -43,8 +43,12 @@ object SparkTransportConf {
    * @param numUsableCores if nonzero, this will restrict the server and client threads to only
    *                       use the given number of cores, rather than all of the machine's cores.
    *                       This restriction will only occur if these properties are not already set.
+   * @param disablePortRetry if true, server will not retry its port. It's better for the long-run
+   *                        server to disable it since the server and client had the agreement of
+   *                        the specific port.
    */
-  def fromSparkConf(_conf: SparkConf, numUsableCores: Int = 0): TransportConf = {
+  def fromSparkConf(_conf: SparkConf, numUsableCores: Int = 0,
```

Comment:
Also, this change broke the MiMA checks. It doesn't feel like these classes should be public (so maybe an exclusion should be fine here), but you can also work around it by declaring an overloaded method instead.

@aarondav any comments about whether these classes are really meant to be public?